Helm-388: Rename Helm in the Helm docs (#653)
OpenNMS/grafana-plugin: docs/modules/reference/pages/how-to-build-with-docker.adoc

= Build from Source with Docker

This section describes how to build the {page-component-title} with Docker.

== Simple build

You can compile the plugin from source with a Docker build image:

. Check out the repository, compile the code, and install the plugin:
+

> [source, console]

Reviewer: ```suggestion
[source, shell]
```

docker build -t mygrafanapluginimage .

. Run the image:
+

> [source, console]

Reviewer: ```suggestion
[source, shell]
```

docker run -p 3000:3000 -t mygrafanapluginimage

. Log in to your Grafana instance with the default admin credentials (`admin`/`admin`).
. Navigate to *Plugins* and enable the *OpenNMS Plugin for Grafana*.

IMPORTANT: The Grafana plugin directory location is changed from `/var/lib/grafana/plugins` to `/opt/grafana/plugins`.
This is because the base Grafana image defines `/var/lib/grafana` as a volume, which cannot be changed.

== Advanced build

> If you want to build a plugin image based on a specific GitHub fork or branch, you can use `--build-arg`:

Reviewer: I think in this case, using the full name makes more sense to reduce ambiguity (i.e. to say we're not referring to _a_ plugin, but _our_ plugin).

```suggestion
If you want to build {page-component-title} based on a specific GitHub fork or branch, you can use `--build-arg`:
```

[source, console]
docker run -p 3000:3000 -t myopennmshelmimage

> . Log in to your Grafana plugin instance with the default admin credentials (`admin`/`admin`).

Reviewer: ```suggestion
. Log in to your Grafana instance with the default admin credentials (`admin`/`admin`).
```

> . Navigate to *Plugins* and enable the *OpenNMS plugin for Grafana*.

Reviewer: ```suggestion
. Navigate to *Plugins* and enable the *OpenNMS Plugin for Grafana*.
```

Horizon and Meridian include templates from which you can create PDF reports.
They use an existing Grafana dashboard (see https://docs.opennms.com/horizon/31/operation/deep-dive/database-reports/templates/grafana.html[Grafana Dashboard] in the main OpenNMS documentation).

> If you want to use this feature, and you are running the plugin through Docker, you must complete additional configuration.

Reviewer: ```suggestion
If you want to use this feature, and you are running the {page-component-title} through Docker, you must complete additional configuration.
```

> Refer to Grafana's https://grafana.com/docs/grafana/latest/administration/image_rendering/#remote-rendering-service[Remote rendering service documentation] for more information.

Reviewer: ```suggestion
Refer to Grafana's https://grafana.com/docs/grafana/latest/administration/image_rendering/#remote-rendering-service[remote rendering service documentation] for more information.
```
